lib/collection/src/collection_builder/collection_builder_base.rs

```diff
@@ -1,137 +0,0 @@
-use std::cmp::max;
-use std::fs::create_dir_all;
-use std::path::Path;
-use std::sync::atomic::{AtomicUsize, Ordering};
-use std::sync::Arc;
-
-use num_cpus;
-use parking_lot::RwLock;
-use segment::payload_storage::schema_storage::SchemaStorage;
-use tokio::runtime;
-use tokio::sync::Mutex;
-
-use segment::segment_constructor::simple_segment_constructor::build_simple_segment;
-use segment::types::HnswConfig;
-
-use crate::collection::Collection;
-use crate::collection_builder::optimizers_builder::build_optimizers;
-use crate::collection_builder::optimizers_builder::OptimizersConfig;
-use crate::collection_manager::holders::segment_holder::SegmentHolder;
-use crate::config::{CollectionConfig, CollectionParams, WalConfig};
-use crate::operations::types::{CollectionError, CollectionResult};
-use crate::operations::CollectionUpdateOperations;
-use crate::update_handler::{Optimizer, UpdateHandler};
-use crate::wal::SerdeWal;
-
-pub fn construct_collection(
```

Comment:
The functions from collection_builder were moved as it seemed to help with readability to have it in an impl block.